### PR TITLE
GB-5538 - Remove db and search

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Breaking
 
- Removes the embedded SQLite database from the CLI (previously deprecated)
- Removes search functionality from the CLI (previously deprecated)

# Type of change

- [x] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

**Comment by obmarg** [comment]
So are we just deleting all the CLI tests?  That doesn't seem ideal...

**Comment by yoav-lavi** [comment]
@obmarg - @fbjork asked to do this, we'll return some of the ones that test features unrelated to the DB around Christmas

**Comment by yoav-lavi** [comment]
@obmarg it's not all of them though, just the ones that use the DB

**Comment by obmarg** [comment]
Have we made a list of the ones we're intending to resurrect?

**Comment by yoav-lavi** [comment]
@obmarg Essentially:



auth
caching
coercion
compilation_error
concurrency?
resolvers
default
watch
home 
length
owner
reserved_dates?
scalars

**Comment by obmarg** [comment]
> @obmarg it's not all of them though, just the ones that use the DB

Sure, but there's some core functionality I see deleted.  `@defer`, the `dev` command, the `start` command, compilation errors, caching

**Comment by yoav-lavi** [comment]
@obmarg yes, all of these currently rely on the DB and we need a local alternative that we can use to replace it for these tests

**Comment by obmarg** [comment]
Could w mark the ones we're planning to resurrect with an `#[ignore]` rather than just deleting?  Seems like it'd make the job of resurrecting them (and reviewing them upon resurrection) easier?

**Comment by yoav-lavi** [comment]
@obmarg They're going to be massively changed, I'm not sure it'd actually make the diff much smaller. But I can look into it if we think it will

https://linear.app/grafbase/issue/GB-5641/restore-the-removed-cli-tests-from-removing-the-db

**Comment by tomhoule** [comment]
Yeah `@ignore` will still require changes because they still have to compile (I think?). Alternatively, we could comment them out.

**Comment by yoav-lavi** [comment]
@obmarg done

File: cli/crates/cli/tests/caching.rs [hunk 1]
```diff
@@ -100,6 +102,8 @@ async fn model_and_field_caching() {
     assert_eq!(header(&response, GRAFBASE_CACHE_HEADER), Some("HIT"));
     assert_eq!(response.headers().typed_get::<CacheControl>(), None);
 }
+
+#[ignore]
```
Comment (yoav-lavi) [inline cli/crates/cli/tests/caching.rs:106]:
```suggestion
#[ignore]
```

**Review by obmarg — APPROVED** [review]
Nice work

**Pull request merged**